Cubical/Structures/Poset.agda

```diff
@@ -0,0 +1,189 @@
+{-# OPTIONS --cubical --safe #-}
+
+module Cubical.Structures.Poset where
+
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Logic
+open import Cubical.Foundations.Isomorphism
+open import Cubical.Foundations.Equiv        hiding   (_■)
+open import Cubical.Foundations.SIP          renaming (SNS-≡ to SNS)
+open import Cubical.Core.Primitives
+open import Cubical.Foundations.HLevels
+open import Cubical.Data.Sigma
+open import Cubical.Data.Sigma.Properties
+
+-- We will adopt the convention of denoting the level of the carrier
+-- set by ℓ₀ and the level of the relation result by ℓ₁.
+private
+  variable
+    ℓ₀ ℓ₁ : Level
+
+Order : (ℓ₁ : Level) → Type ℓ₀ → Type (ℓ-max ℓ₀ (ℓ-suc ℓ₁))
+Order ℓ₁ A = A → A → hProp ℓ₁
+
+order-iso : (M N : Σ (Type ℓ₀) (Order ℓ₁)) →  fst M ≃ fst N → Type (ℓ-max ℓ₀ ℓ₁)
+order-iso (A , _⊑₀_) (B , _⊑₁_) eqv =
+  (x y : A) → [ x ⊑₀ y ⇔ f x ⊑₁ f y ]
+  where
+    f = equivFun eqv
```

Comment:
Nitpicking: instead of using equivFun eqv in a where you can just pattern match on eqv. That way this would be a oneliner